typescript/src/bitcoin.ts

```diff
@@ -373,19 +374,20 @@ export function computeHash160(text: string): string {
  *        unprefixed hex string (without 0x prefix).
  * @param witness - If true, a witness public key hash will be encoded and
  *        P2WPKH address will be returned. Returns P2PKH address otherwise
- * @param network - Network that the address should be encoded for.
- *        For example, `main` or `testnet`.
+ * @param network - Network the address should be encoded for.
  * @returns P2PKH or P2WPKH address encoded from the given public key hash
+ * @throws Throws an error if network is not supported.
  */
 export function encodeToBitcoinAddress(
   publicKeyHash: string,
   witness: boolean,
-  network: string
+  network: BitcoinNetwork
```

Comment:
There are more places where we expect `network: string` (e.g `calculateDepositAddress`). Shouldn't we replace them with `network: BitcoinNetwork` as well?

Reply:
Good catch! 435e93ae2735bcfdb08da9b69d72fed988a36430

I can't find any other places though.